hack/tools/runtime-openapi-gen/main.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"os"
+	"path"
+
+	flag "github.com/spf13/pflag"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	"sigs.k8s.io/cluster-api/internal/runtime/catalog"
+)
+
+var (
+	version    = flag.String("version", "", "Version for the OpenAPI specification.")
+	outputFile = flag.String("output-file", "runtime-sdk-openapi.yaml", "Output file name.")
+)
+
+func main() {
+	flag.Parse()
+
+	if *version == "" {
+		klog.Exit("--version must be specified")
+	}
+	if *outputFile == "" {
+		klog.Exit("--output-file must be specified")
+	}
+
+	outputFileExt := path.Ext(*outputFile)
+	if outputFileExt != ".yaml" && outputFileExt != ".json" {
+		klog.Exit("--output-file must have either 'yaml' or 'json' extension")
+	}
+
+	c := catalog.New()
+	_ = runtimehooksv1.AddToCatalog(c)
+
+	// TODO(openapi): TBD if we want to support types from "external" packages which only have GetOpenAPIDefinitions.
+	// If yes, we should add an additional flag for those.
+	// Follow-up: If we don't want to support it drop openapi generation in api/v1beta1.
+	c.AddOpenAPIDefinitions(clusterv1.GetOpenAPIDefinitions)
```

Comment:
Looks like changes in catalog + openapi.go are missing

Reply:
ack 🤦